src/dune_rules/ml_sources.ml

```diff
@@ -61,7 +61,10 @@ module Modules = struct
           libs
           ~init:(Lib_name.Set.empty, Lib_info.Sentinel.Map.empty)
           ~f:(fun (lib_set, acc) part ->
-            let name = Library.best_name part.stanza in
+            (* we need to check for private name to avoid "multiple rules" errors,
+               because even for public libraries, the artifacts folder still uses
+               the private name *)
+            let name = Library.private_name part.stanza in
```

Comment:
I wonder if we should be checking directly what conflicts. In this case, check the obj_dir path equality rather than the private name equality?

Reply:
I do actually prefer the current version than the previous one. This looks way better IMO